### PR TITLE
Feature/new syscall tests cases

### PR DESCRIPTION
**Pull request by kpan2034**
Added tests for the following syscalls:

- pipe2
- getifaddrs
- getppid
- mkdir
- rmdir
- access

### CODE REVIEW

**Comment by rennergade** [comment]
We should hook these up to the test suite before we merge them. I'll let the team review the individual tests for now.

**Review by RusherRG — CHANGES_REQUESTED** [review]

File: tests/test_cases/getppid.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#undef _GNU_SOURCE
```
[the remark this answers is not in the document]
Comment (rennergade) [reply tests/test_cases/getppid.c:1]: Yes definitely.

**Comment by rennergade** [comment]
Also naming the test cases that handle two syscalls so we know that off the bat would be good. IE "access" should be "creat_access"

File: tests/test_cases/getppid.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (kpan2034) [reply tests/test_cases/getppid.c:1]: Alright, will work on this.

[the remark this answers is not in the document]
Comment (RusherRG) [reply tests/test_cases/getppid.c:1]: Awesome, this looks good. :tada:

File: tests/test_cases/pipe2.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        exit(-1)
+
+# Nondeterministic digit check
+lind_line0 = lind_result[0].split(',')
+host_line0 = host_result[0].split(',')
+
+if not (lind_line0[0][-1].isdigit() and lind_line0[1][1].isdigit() \
+    and host_line0[0][-1].isdigit() and host_line0[1][1].isdigit()):
+    print "Nondeterministic lines of an unrecognized format!"
+    exit(-1)
```
Comment (RusherRG) [inline tests/test_cases/pipe2.py:27]: Make sure to end all your test files with a new line.

File: tests/test_cases/getifaddrs.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    if not (tokens[2][1:-1].isdigit() and tokens[2][0] == '(' and tokens[2][-1] == ')'):
+        print "Expected the last token to be numeric"
+        exit(-1)
+
+for line in lindoutput:
+    verify(line)
+
+for line in hostoutput:
+    verify(line)
```
Comment (RusherRG) [inline tests/test_cases/getifaddrs.py:34]: Make sure to end all your test files with a new line.

File: tests/test_cases/getppid.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      exit(EXIT_FAILURE);
+    }
+  }
+  // Parent process
+  else {
+    // Wait till child process returns, and then only then exit
+    wait(NULL);
+  }
+  return 0;
+}
```
Comment (RusherRG) [inline tests/test_cases/getppid.c:43]: Make sure to end all your test files with a new line.

**Review by RusherRG — CHANGES_REQUESTED** [review]

File: tests/test_cases/pipe2.py
Comment (RusherRG) [inline tests/test_cases/pipe2.py:1]: Can you also add checks for the return value of read and write? Both of them should be 3(number of bytes read/write)

**Review by RusherRG — CHANGES_REQUESTED** [review]

File: tests/test_cases/getppid.c [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+int main(void) {
+
+  // Get pid of current process
+  pid_t pid = getpid();
+  fflush(stdout);
```
Comment (kuzeyardabulut) [inline tests/test_cases/getppid.c:14]: Why we are using `fflush()` in here ?

**Comment by kuzeyardabulut** [comment]
You should `fflush(stdout)` after doing printfs.

**Review by kuzeyardabulut — CHANGES_REQUESTED** [review]
You should `fflush(stdout)` after doing printfs.

File: tests/test_cases/pipe2.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+# Verify # of bytes that write and read return
+lines = [lind_result[1].split(), host_result[1].split(), lind_result[2].split(), host_result[2].split()]
+for line in lines:
+    if not (line[-1] == '3'):
+        print "Incorrect number of bytes written!"
+        exit(-1)
```
Comment (rennergade) [inline tests/test_cases/pipe2.py:34]: needs newline

File: tests/test_cases/pipe2.py
Comment (rennergade) [reply tests/test_cases/pipe2.py:1]: that would be good

File: tests/test_cases/pipe2.c [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    printf("write(): %s\n", strerror(errno));
+    exit(EXIT_FAILURE);
+  }
+  printf("write() ret: %d\n", ret);
+
+  if ((ret = read(fd[0], str, 3)) < 0) {
+    printf("read(): %s\n", strerror(errno));
+    exit(EXIT_FAILURE);
+  }
+  printf("read() ret: %d\n", ret);
```
Comment (rennergade) [inline tests/test_cases/pipe2.c:32]: add fflushes

File: tests/test_cases/creat_access.c [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  if (access_result == -1) {
+    perror("INVALID PERMISSIONS FOR FILE");
+    exit(EXIT_FAILURE);
+  }
+  printf("File has the right permissions\n");
+  fflush(stdout);
+
+  // Cleanup
+  remove(FILE);
+  fflush(stdout);
```
Comment (kuzeyardabulut) [inline tests/test_cases/creat_access.c:44]: Do we need this?

File: tests/test_cases/getppid.c [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+#include <sys/wait.h>
+#include <unistd.h>
+
+int main(void) {
+
+  // Get pid of current process
+  pid_t pid = getpid();
+
+  // Fork
+  int f = fork();
```
Comment (kuzeyardabulut) [inline tests/test_cases/getppid.c:16]: I think it would be better to delete such short comment lines.

**Review by kuzeyardabulut — CHANGES_REQUESTED** [review]

**Pull request merged**